### PR TITLE
#8520: Reduce dashboards clutter and widget update

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR attempts to declutter the widgets in dashboard to allow more widgets to be viewed. Also provides a possibility to configure widgets options (especially `DataGrid` options) through dashboard plugin

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
#8520 

**What is the new behavior?**
New dashboard (100%)
2022-09-06 13_43_07-MapStore HomePage

Map viewer
2022-09-06 13_42_05-DevTools - localhost_8081__debug=true

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
Ref: https://github.com/geosolutions-it/MapStore2-C040/issues/495

### CODE REVIEW

File: web/client/plugins/Dashboard.jsx [hunk 1]
```diff
@@ -125,13 +125,30 @@ const WidgetsView = compose(
  * @prop {object} cfg.cols Number of columns in this layout. default { lg: 6, md: 6, sm: 4, xs: 2, xxs: 1 }
  * @prop {object} cfg.minLayoutWidth minimum size of the layout, below this size the widgets are listed in a single column
  * for more info about rowHeight and cols, see https://github.com/STRML/react-grid-layout#grid-layout-props
+ * @prop {object} cfg.widgetOpts widget specific options
```
Comment (MV88) [inline web/client/plugins/Dashboard.jsx:128]: which options can contain gridOpts ?
let's put more details here

File: web/client/plugins/Dashboard.jsx [hunk 2]
```diff
@@ -125,13 +125,30 @@ const WidgetsView = compose(
...
+ * @example
+ * {
+ *   "name": "Dashboard",
+ *   "cfg": {
+ *      "rowHeight": 150,
+ *      "cols": { lg: 6, md: 6, sm: 4, xs: 2, xxs: 1 },
+ *      "widgetOpts": {
+ *          "table": {
```
Comment (MV88) [inline web/client/plugins/Dashboard.jsx:136]: can you list somewhere above the list of entries that can be used ? table, counter etc.

File: web/client/themes/default/less/widget.less [hunk 3]
```diff
@@ -9,6 +9,12 @@
 // **************
 // Theme
 // **************
+.setFontSize (@val){
+    @fSize: @val
+};
+.em (@prop, @val) {
```
Comment (MV88) [inline web/client/themes/default/less/widget.less:15]: what if we want to use this function somewhere else?
is there a way to include it in the mapstore theme?
like this color-var in css-properties.less?

https://github.com/geosolutions-it/MapStore2/blob/7567d12b49fea8a00ac827b77708856e9927b125/web/client/themes/default/less/mixins/css-properties.less#L15-L21

**Review by MV88 — CHANGES_REQUESTED** [review]
looks nice overall. but I noticed some things that we can discuss together if they have to done or not

- [x] the filter field **width** inside table widget when there is space remains small

DEV
image

PR
image

- [ ] second point, the header of map widgets seems to have a different height,
image
Probably it is because the map widget filter
image

- [ ] the title of widgets before it was centered, now it is left aligned
image

**Comment by MV88** [comment]
@dsuren1 I've spoken with @tdipisa.

let's ignore for now second and third points and the thing related to css-properties.less?
@allyoucanmap please give us some insight when you are back

**Comment by tdipisa** [comment]
@MV88 a few clarifications below:

 >the filter field width inside table widget when there is space remains small

To be fixed

>second point, the header of map widgets seems to have a different height,

@dsuren1 let me know if we can do something for this relatively quickly otherwise leave it as it is for now

>the title of widgets before it was centered, now it is left aligned

This is coming from the @allyoucanmap proposal therefore leave it as it is for now.

**Comment by dsuren1** [comment]
@tdipisa 
> > second point, the header of map widgets seems to have a different height,
> 
> @dsuren1 let me know if we can do something for this relatively quickly otherwise leave it as it is for now

Yes, can be fixed. Working on it.

**Comment by dsuren1** [comment]
Fixed
**Map widget (Map switcher)**
2022-09-06 17_01_28-MapStore HomePage

**Map widget (one map)**
2022-09-06 17_02_00-MapStore HomePage

**Review by MV88 — APPROVED** [review]
A bit better on second point, at leas in case of one map only 
LGTM

**Comment by MV88** [comment]
@tdipisa let us know when we can merge this

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV and let us know. For the backport we have to wait after the release of 2022.02.00

**Pull request merged**